geom/src/polygon.rs

```diff
@@ -7,51 +7,41 @@ use serde::{Deserialize, Serialize};
 use abstutil::Tags;
 
 use crate::{
-    Angle, Bounds, CornerRadii, Distance, GPSBounds, HashablePt2D, LonLat, PolyLine, Pt2D, Ring,
-    Tessellation, Triangle,
+    Angle, Bounds, CornerRadii, Distance, GPSBounds, LonLat, PolyLine, Pt2D, Ring, Tessellation,
+    Triangle,
 };
 
 #[derive(PartialEq, Serialize, Deserialize, Clone, Debug)]
 pub struct Polygon {
-    pub(crate) points: Vec<Pt2D>,
-    /// Groups of three indices make up the triangles
-    pub(crate) indices: Vec<u16>,
-
-    /// If the polygon has holes, explicitly store all the rings (the one outer and all of the
-    /// inner) so they can later be used to generate outlines and such. If the polygon has no
-    /// holes, then this will just be None, since the points form a ring.
-    rings: Option<Vec<Ring>>,
+    // [0] is the outer/exterior ring, and the others are holes/interiors
+    pub(crate) rings: Vec<Ring>,
+    // For performance reasons, some callers may want to generate the polygon's Rings and
```

Comment:
There is only one use of this -- `PolyLine`. I still have problems doing profiling on my Linux setup. When I sort that out, I'll look into just using `earcutr` for those cases too.